xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -674,53 +674,77 @@ absl::Status IrEmitterUnnested::EmitCublasLtMatmulThunk(
   xla::gpu::GemmBackendConfig config = gpu_config.gemm_backend_config();
   xla::gpu::GemmBackendConfig_Epilogue epilogue = config.epilogue();
 
+  TF_ASSIGN_OR_RETURN(bool is_forward_mode,
+                      xla::gpu::gpublas_lt::EpilogueForForward(epilogue));
   TF_ASSIGN_OR_RETURN(bool has_vector_bias,
                       xla::gpu::gpublas_lt::EpilogueAddsVectorBias(epilogue));
   bool has_matrix_bias = config.beta() != 0;
-
-  TF_RET_CHECK(instr->operand_count() ==
-               2 + int{has_matrix_bias} + int{has_vector_bias});
-
   TF_ASSIGN_OR_RETURN(
       bool has_aux_output,
       xla::gpu::gpublas_lt::EpilogueHasAuxiliaryOutput(epilogue));
-  xla::ShapeIndex output_index =
-      has_aux_output ? xla::ShapeIndex{0} : xla::ShapeIndex{};
+  TF_ASSIGN_OR_RETURN(
+      bool has_aux_input,
+      xla::gpu::gpublas_lt::EpilogueHasAuxiliaryInput(epilogue));
+  if (is_forward_mode) {
+    TF_RET_CHECK(instr->operand_count() ==
+                 2 + int{has_matrix_bias} + int{has_vector_bias});
+  } else {
+    TF_RET_CHECK(instr->operand_count() == 2 + int{has_aux_input});
```

Comment:
Is it possible to have both a bias and an aux input? Admittedly this would be unusual since we only expect aux inputs on the backwards pass, which typically doesn't have bias adds

Reply:
I cannot think of one case.